reference-implementation/lib/readable-byte-stream.js

```diff
@@ -0,0 +1,965 @@
+  }
+
+  cancel(reason) {
+    if (!IsReadableByteStreamByobReader(this)) {
+      return Promise.reject(
+        new TypeError(
+            'ReadableByteStreamByobReader.prototype.cancel can only be used on a ReadableByteStreamByobReader'));
+    }
+
+    if (this._state === 'closed') {
+      return Promise.resolve(undefined);
+    }
+
+    if (this._state === 'errored') {
+      return Promise.reject(this._storedError);
+    }
+
+    assert(this._ownerReadableByteStream !== undefined, 'This stream must be attached to a stream');
+
+    return CancelReadableByteStream(this._ownerReadableByteStream, reason);
+  }
+
+  read(view) {
+    if (!IsReadableByteStreamByobReader(this)) {
+      return Promise.reject(
+        new TypeError(
+            'ReadableByteStreamByobReader.prototype.read can only be used on a ReadableByteStreamByobReader'));
+    }
+
+    if (view === undefined || !ArrayBuffer.isView(view)) {
+      return Promise.reject(new TypeError('Valid view must be provided'));
+    }
+
+    const ctor = view.constructor;
+    let elementSize = 1;
+    if (ctor === Int16Array || ctor === Uint16Array || ctor === Int32Array || ctor === Uint32Array ||
+        ctor === Float32Array || ctor === Float64Array || ctor === Int8Array || ctor === Uint8Array ||
+        ctor === Uint8ClampedArray) {
+      elementSize = ctor.BYTES_PER_ELEMENT;
+    } else if (ctor !== DataView) {
+      return Promise.reject(new TypeError('view is of an unsupported type'));
+    }
+
+    if (view.byteLength === 0) {
+      return Promise.reject(new TypeError('view must have non-zero byteLength'));
+    }
+
+    if (this._state === 'errored') {
+      assert(this._ownerReadableByteStream === undefined, 'This reader must be detached');
+
+      return Promise.reject(this._storedError);
+    }
+
+    if (this._state === 'closed' && this._ownerReadableByteStream === undefined) {
+      return Promise.resolve(CreateIterResultObject(new ctor(view.buffer, view.byteOffset, 0), true));
+    }
+
+    const promise = new Promise((resolve, reject) => {
+      const req = {
+        resolve,
+        reject,
+        byteOffset: view.byteOffset,
+        byteLength: view.byteLength,
+        ctor,
+        elementSize
+      };
+      this._readIntoRequests.push(req);
+    });
+
+    PullFromReadableByteStreamInto(
+        this._ownerReadableByteStream, view.buffer, view.byteOffset, view.byteLength, elementSize);
+
+    return promise;
+  }
+
+  releaseLock() {
+    if (!IsReadableByteStreamByobReader(this)) {
+      throw new TypeError(
+          'ReadableByteStreamByobReader.prototype.releaseLock can only be used on a ReadableByteStreamByobReader');
+    }
+
+    if (this._ownerReadableByteStream === undefined) {
+      return;
+    }
+
+    if (this._readIntoRequests.length > 0) {
+      throw new TypeError('Tried to release a reader lock when that reader has pending read() calls un-settled');
+    }
+
+    if (this._ownerReadableByteStream._state === 'readable') {
```

Comment:
We have the auto release feature. So, this should be assert(). Fixed by f3be413272042e204083abbf52aa875e9cd753bc.